System.Device.Gpio/GpioController.cs

```diff
@@ -0,0 +1,268 @@
+        }
+
+        private void GetController()
+        {
+            if (_syncLock == null)
+            {
+                _syncLock = new object();
+            }
+
+            lock (_syncLock)
+            {
+                _controller = Windows.Devices.Gpio.GpioController.GetDefault();
+                if (_gpioPins == null)
+                {
+                    _gpioPins = new Windows.Devices.Gpio.GpioPin[_controller.PinCount];
+                    _gpioEvents = new PinEventTypes[_controller.PinCount];
+                    _gpioEventsHappening = new PinEventTypes[_controller.PinCount];
+                    _gpioPinChange = new PinChangeEventHandler[_controller.PinCount];
+                }
+            }
+        }
+
+        /// <summary>
+        /// The numbering scheme used to represent pins provided by the controller.
+        /// </summary>
+        public PinNumberingScheme NumberingScheme { get; internal set; }
+
+        /// <summary>
+        /// The number of pins provided by the controller.
+        /// </summary>
+        public int PinCount => _controller.PinCount;
+
+        /// <summary>
+        /// Closes an open pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        public void ClosePin(int pinNumber)
+        {
+
+            if (_gpioPins[pinNumber] != null)
+            {
+                _gpioPins[pinNumber].Dispose();
+                _gpioPins[pinNumber] = null;
+            }
+            else
+            {
+                throw new IOException($"Port {pinNumber} is not open");
+            }
+        }
+
+        /// <summary>
+        /// Dispose the controller
+        /// </summary>
+        public void Dispose()
+        {
+            for (int i = 0; i < _gpioPins.Length; i++)
+            {
+                ClosePin(i);
+            }
+        }
+
+        /// <summary>
+        /// Gets the mode of a pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        /// <returns>The mode of the pin.</returns>
+        public PinMode GetPinMode(int pinNumber)
+        {
+            if (_gpioPins[pinNumber] == null)
+            {
+                throw new IOException($"Port {pinNumber} is not open");
+            }
+
+            // It is safe to cast, enums are the same
+            return (PinMode)_gpioPins[pinNumber].GetDriveMode();
+        }
+
+        /// <summary>
+        /// Checks if a pin supports a specific mode.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        /// <param name="mode">The mode to check.</param>
+        /// <returns>The status if the pin supports the mode.</returns>
+        public bool IsPinModeSupported(int pinNumber, PinMode mode) => _gpioPins[pinNumber].IsDriveModeSupported((Windows.Devices.Gpio.GpioPinDriveMode)mode);
```

Comment:
Same here, similar check (which is not here) and a `IsDriveModeSupported(pinNumber, mode)`.

OK to have it this way for now.